storjscandb/database.go

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2021 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storjscandb
+
+import (
+	"context"
+
+	"github.com/spacemonkeygo/monkit/v3"
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/private/dbutil"
+	"storj.io/private/dbutil/pgutil"
+	"storj.io/private/migrate"
+	"storj.io/private/tagsql"
+	"storj.io/storjscan/storjscandb/dbx"
+)
+
+var (
+	mon = monkit.Package()
+
+	// Error is the default storjscandb errs class.
+	Error = errs.Class("storjscandb")
+)
+
+// DB is storjscan database.
+type DB struct {
+	*dbx.DB
+	log            *zap.Logger
+	driver         string
+	source         string
+	implementation dbutil.Implementation
+	migrationDB    tagsql.DB
+}
+
+// Open creates instance of storjscan DB.
+func Open(ctx context.Context, log *zap.Logger, databaseURL string) (*DB, error) {
+	driver, source, impl, err := dbutil.SplitConnStr(databaseURL)
+	if err != nil {
+		return nil, err
+	}
+	if impl != dbutil.Postgres && impl != dbutil.Cockroach {
```

Comment:
the check for supported drivers seems to imply that either a postgres or cockroach impl is allowed, but below in the migration to latest schema, only postgres impl performs the migration.

Reply:
Disabled cockroach